pkg/admission/cert/writer/fs_test.go

```diff
@@ -0,0 +1,356 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package writer
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"sync/atomic"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	admissionregistration "k8s.io/api/admissionregistration/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/admission/cert/generator"
+	fakegenerator "sigs.k8s.io/controller-runtime/pkg/admission/cert/generator/fake"
+)
+
+var _ = Describe("FSCertWriter", func() {
+	var mwc *admissionregistration.MutatingWebhookConfiguration
+	var vwc *admissionregistration.ValidatingWebhookConfiguration
+	var url string
+	var count uint64 = 0
+
+	BeforeEach(func(done Done) {
+		atomic.AddUint64(&count, 1)
+		url = "https://example.com/admission"
+		mwc = &admissionregistration.MutatingWebhookConfiguration{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "admissionregistration.k8s.io/v1beta1",
+				Kind:       "MutatingWebhookConfiguration",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        "test-mwc",
+				Namespace:   "test-ns",
+				UID:         types.UID("123456"),
+				Annotations: map[string]string{},
+			},
+			Webhooks: []admissionregistration.Webhook{
+				{
+					Name: "webhook-1",
+					ClientConfig: admissionregistration.WebhookClientConfig{
+						Service: &admissionregistration.ServiceReference{
+							Namespace: "test-svc-namespace",
+							Name:      "test-service",
+						},
+						CABundle: []byte(``),
+					},
+				},
+			},
+		}
+		vwc = &admissionregistration.ValidatingWebhookConfiguration{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "admissionregistration.k8s.io/v1beta1",
+				Kind:       "MutatingWebhookConfiguration",
```

Comment:
`vwc` is declared to be validating web hook configuration, but the `Kind` filed is mutating webhook configuration.

Reply:
Good catch!